app/controllers/authentication_controller.rb

```diff
@@ -39,4 +32,8 @@ def login_error(message)
     flash[:notice] = message
     redirect_to_login
   end
+
+  def load_user
+    GithubActions::LoadUserService.new(params[:user][:token]).perform
```

Comment:
this method used three times, pls memoize it or use `:before_action` for loading user into instance variable.